corpus/classes.txt

```diff
@@ -55,16 +58,18 @@ class Strings {
 ---
 
 (source_file
-	(class_declaration (type_identifier)
-		(class_body
-			(function_declaration (simple_identifier) (function_body
-				(line_string_literal)))
-			(function_declaration (simple_identifier) (function_body
-				(additive_expression
-					(additive_expression
-						(line_string_literal)
-						(line_string_literal))
-					(line_string_literal)))))))
+	(class_declaration identifier: (type_identifier)
+		class_body: (class_body
+                        (function_declaration identifier: (simple_identifier)
+                            function_body: (function_body
+                                expression: (line_string_literal)))
```

Comment:
`function_body` is a little bit complicate actually(maybe the most complicate part, because it mapping to `block/statement` and `expression` at same time), so still use a normal rule for it.

```
		function_body: $ => choice(
		    $._block,
		    seq("=", field("expression", $._expression))
		),
```

Reply:
Then I would inline `_function_body`, extract a `expression_body` grammar rule `seq("=", field("expression", $._expression))` to have a parse tree node for that, and de-inline `_block`. This would allow to discriminate between a regular block and a single expression block by node type instead of testing for the presence of an `expression` field within a `choice` rewriter.

Also on the AST side, I guess we could add a `Block` and an `ExpressionBlock`.

So on the rewriting side, we would have:
```
def block(self):
    return Block(statements=_.children_)

def expression_block(self):
    return ExpressionBlock(expression=_.expression)
```

instead of:

```
def function_body(self):
    return choice(
        _(expression=_) >> ... expression block ...,
       ... regular block ...
    )
```

But again, not necessarily for this PR, but to keep in the back of our heads :)